infra/aws/terraform/prow-build-cluster/README.md

```diff
@@ -243,3 +243,31 @@ make destroy
 
 If you want to remove roles used for EKS creation go to `../iam/<aws_account_name>` and run `terraform destroy` command there.
 
+
+## GitOps
+
+To sync state from git repo into EKS cluster we use [FluxCD](https://fluxcd.io/).
+
+Flux Kustomizations and HelmReleases for manifests are stored inside the `./resources` directory and have been generated with use of `./hack/flux-update.bash`. On top of that, the script prepares manifests for the GitOps Tool Kit and can be used to update its version.
+
+The `flux-system` namespace contains all GitOps Tool Kit componenets as well as all Flux Kustomizations.
+
+Flux Helm Releases ought to be deployed in the same namespaces as manifests they create. As a convention, all Flux Helm Releases have to be prefixed with `flux-hr-`. This convention is leveraged by automation.
+
+#### Installing Flux
+
+```bash
+make flux-install
+```
+
+#### Deploying Kustomizations
+
+```bash
+make flux-apply
+```
+
+#### Deploying Helm Releases
+
+```bash
+make flux-apply-helm
+```
```

Comment:
How does this affect other parts of this document? If we have Flux to apply everything in `resources`, I don't think that we need most of phase 4 (and it might be outdated too).

Reply:
`make flux-apply` applies only kostumizations that are inside `resources` folder. HelmReleases are managed independently, hence we need to apply them separately. To avoid confusion I'll rename `flux-apply` to `flux-apply-kustomizations`.

Reply:
I'm not sure I follow. Kustomizations inside the `resources` folder are used to reconcile those resources. If those resources don't exist, Flux is going to deploy them upon reconciliation. That's at least my understanding, please correct me if I'm wrong.

If my understanding is correct, phase 4 is not needed at all because Flux is going to deploy all those resources, you don't need and shouldn't deploy them manually. The only thing you should deploy manually if Flux.

Reply:
Yeah, you are right. The only requirement is that Sources and Helm Releases should be part of Kustomizations, and that's what we are doing. Let me remove redundant docs.